x/programs/examples/imports/program/program.go

```diff
@@ -59,12 +59,15 @@ func (i *Import) Register(link *host.Link, callContext program.Context) error {
 }
 
 // callProgramFn makes a call to an entry function of a program in the context of another program's ID.
-func (i *Import) callProgramFn(callContext program.Context) func(*wasmtime.Caller, int64, int64, int64, int64) int64 {
+func (i *Import) callProgramFn(callContext program.Context) func(*wasmtime.Caller, int32, int32, int32, int32, int32, int32, int64) int64 {
 	return func(
 		wasmCaller *wasmtime.Caller,
-		programID int64,
-		function int64,
-		args int64,
+		programPtr int32,
+		programLen int32,
+		functionPtr int32,
+		functionLen int32,
+		argsPtr int32,
+		argsLen int32,
```

Comment:
Instead of casting all of these, I think you can just make them `uint32` to begin with, can't you? 

Don't worry about it for this PR, but see if you can avoid casting when tackle the rest of #862

Reply:
I don't think that it's something that works in WASM, the only supported primitive types are `i32`, `i64`, `f32`, `f64`

Reply:
I'm getting the following error: "function types incompatible: expected func of type `(i32, i32, i32, i32, i32, i32, i64) -> (i64)`, found func of type `(externref, externref, externref, externref, externref, externref, i64) -> (i64)"

Reply:
That's super weird. When using `wasmtime` directly in Rust, the only part of the type check that matters is the size. I just tried it out on the `public_functions.rs` code, and I swapped the `allocate` type signature to use `u32` instead of `i32`, and it worked exactly as expected without changing the actual wasm.

Reply:
Indeed, I just swapped the function parameters of the `call_program` linked function, I don't think that there is anything else to do ?